src/utils/rename.js

```diff
@@ -33,3 +35,21 @@ export default function (options) {
     done();
   };
 }
+
+export function hooks (options) {
+  return function rename (files, metalsmith, done) {
+    each(
+      Object.keys(files),
+      function (file, next) {
+        if (match(file, ['*.js']).length) {
+          let newName = file.replace('hook', options.name);
+          debug(`Renaming template ${file} to ${newName}`);
+          files[newName] = files[file];
+          delete files[file];
+        }
+      }
+    );
+
+    done();
+  };
```

Comment:
Same here, the duplication can be minimized, but will wait until I've landed happy path on all generations and then refactor with the full picture in view.